### PR TITLE
[14.0][IMP] stock_request_purchase: Propagate the cancellation of related purchases

### PR DESCRIPTION
**Pull request by victoralmau**
Propagate the cancellation of related purchases

Please @pedrobaeza and @pilarvargas-tecnativa can you review it?

@Tecnativa TT43414

### CODE REVIEW

File: stock_request_purchase/models/stock_request.py [hunk 1]
```diff
@@ -38,6 +39,13 @@ def _check_purchase_company_constrains(self):
                 )
             )
 
+    def action_cancel(self):
+        res = super().action_cancel()
```
Comment (pedrobaeza) [inline stock_request_purchase/models/stock_request.py]: Remember to put docstrings:

```suggestion
    def action_cancel(self):
        """Propagate the cancellation to the generated purchase orders."""
        res = super().action_cancel()
```

**Review by pedrobaeza — APPROVED** [review]
Thanks for adding extra tests for the use case.

**Review by pilarvargas-tecnativa — APPROVED** [review]
thanks :+1:

**Pull request merged**